### PR TITLE
Travis: fix failing tests

### PR DESCRIPTION
**Pull request by Slamdunk**
So, here we are. Comments on code.

### CODE REVIEW

File: src/Message.php [hunk 1]
```diff
@@ -157,7 +157,10 @@ public function isDraft()
      */
     public function isSeen()
     {
-        return 'U' != $this->getHeaders()->get('unseen');
+        return (
+                'R' === $this->getHeaders()->get('recent')
+            ||  ('' === $this->getHeaders()->get('recent') &&  '' !== $this->getHeaders()->get('unseen'))
+        );
```
Comment (Slamdunk) [inline src/Message.php]: References:

1. https://secure.php.net/manual/en/function.imap-headerinfo.php
1. https://github.com/php/php-src/blob/master/ext/imap/php_imap.c#L2124-L2125

File: src/Message/Headers.php [hunk 2]
```diff
@@ -48,7 +48,7 @@ private function parseHeader($key, $value)
             case 'draft':
                 // no break
             case 'unseen':
-                return (bool)trim($value);
+                return $value;
```
Comment (Slamdunk) [inline src/Message/Headers.php]: Flags can't be force to boolean because they can have multiple values, like `recent`:

https://secure.php.net/manual/en/function.imap-headerinfo.php

File: src/Message/Part.php [hunk 3]
```diff
@@ -179,14 +179,14 @@ public function getDecodedContent($keepUnseen = false)
...
-                );
-            }
+            // if ($this->getType() === self::TYPE_TEXT
+                // && strtolower($this->getCharset()) != 'utf-8'
+            // ) {
+                // $this->decodedContent = Transcoder::create()->transcode(
+                    // $this->decodedContent,
+                    // $this->getCharset()
+                // );
+            // }
```
Comment (Slamdunk) [inline src/Message/Part.php]: After commenting this code no test fails, and [testEncoding7Bit](https://github.com/ddeboer/imap/blob/master/tests/MessageTest.php#L41) passes: @ddeboer do you have any test case that justify this piece of code?

File: src/Parameters.php [hunk 4]
```diff
@@ -36,9 +36,10 @@ protected function decode($value)
         $decoded = '';
         $parts = imap_mime_header_decode($value);
         foreach ($parts as $part) {
-            $charset = 'default' == $part->charset ? 'auto' : $part->charset;
+            // $charset = 'default' == $part->charset ? 'auto' : $part->charset;
             // imap_utf8 doesn't seem to work properly, so use Transcoder instead
-            $decoded .= Transcoder::create()->transcode($part->text, $charset);
+            // $decoded .= Transcoder::create()->transcode($part->text, $charset);
+            $decoded .= $part->text;
```
Comment (Slamdunk) [inline src/Parameters.php]: Like Message\Part also headers are affected: when the raw value is used, no test fails and [testEncoding7Bit](https://github.com/ddeboer/imap/blob/master/tests/MessageTest.php#L41) passes. @ddeboer do you have any test case that justify this piece of code?

File: tests/MessageTest.php [hunk 5]
```diff
@@ -101,6 +102,7 @@ public function testDelete()
 
         $message = $this->mailbox->getMessage(3);
         $message->delete();
+        $this->mailbox->expunge();
```
Comment (Slamdunk) [inline tests/MessageTest.php]: Trivial bugfix. [Manual](https://secure.php.net/manual/en/function.imap-delete.php) says:

> Marks messages listed in msg_number for deletion. Messages marked for deletion will stay in the mailbox until either imap_expunge() is called or imap_close() is called with the optional parameter CL_EXPUNGE.

File: src/Message/Part.php (hunk 3 above)
Comment (ddeboer) [reply src/Message/Part.php]: This was a fix for #3, so also `testSubjectEncoding()`.

Comment (Slamdunk) [reply src/Message/Part.php]: I don't know what changed since november 2013, but the is that if you removing code doesn't break tests, it's either:

1. the tests are not enough
1. the code is unnecessary

But don't be afraid: we need a lot of test cases with different charset to test against, and I will gather and test them, in future PR, of couse before the any next release.

Just not now: before deeping into the charset topic, we need minor improvement like coding standards and re-enable code-coverage; one step at time

File: tests/MessageTest.php (hunk 5 above)
Comment (ddeboer) [reply tests/MessageTest.php]: If we only fix this here, this may not be transparent to users. Should we also add a doc comment to `Message::delete()`?

File: src/Message/Part.php (hunk 3 above)
Comment (ddeboer) [reply src/Message/Part.php]: But if the tests pass without this, let’s remove it.

File: tests/MessageTest.php [hunk 6]
```diff
@@ -48,11 +48,12 @@ public function testEncodingQuotedPrintable()
         $raw = "Subject: ESPAÑA\r\n"
             . "Date: =?ISO-8859-2?Q?Fri,_13_Jun_2014_17:18:44_+020?= =?ISO-8859-2?Q?0_(St=F8edn=ED_Evropa_(letn=ED_=E8as))?=\r\n"
             . "Content-Type: multipart/alternative; boundary=\"$boundary\"\r\n"
+            . "\r\n"
             . "--$boundary\r\n"
             . "Content-Transfer-Encoding: quoted-printable\r\n"
             . "Content-Type: text/html; charset=\"windows-1252\"\r\n"
             . "\r\n"
-            . "<html><body>Espa=F1a</body></html>\r\n\r\n"
+            . "<html><body>Espa=C3=B1a</body></html>\r\n\r\n"
```
Comment (ddeboer) [inline tests/MessageTest.php]: Why did you change this? The original text was received in a bug report.

File: src/Parameters.php (hunk 4 above)
Comment (ddeboer) [reply src/Parameters.php]: No, so let’s disable it for now.

File: src/Message/Part.php (hunk 3 above)
Comment (Slamdunk) [reply src/Message/Part.php]: Right: I hate commented code too and for sure will be removed soon.
One step at time, reminder added: https://github.com/ddeboer/imap/issues/174

File: tests/MessageTest.php (hunk 5 above)
Comment (Slamdunk) [reply tests/MessageTest.php]: So, since `delete` doesn't delete the message, I suggest:

1. Rename `delete` method to `markAsDelete`
1. Add to the doc that `mailbox::expunge` must be called for a complete delete

Reminder: https://github.com/ddeboer/imap/issues/175

File: tests/MessageTest.php (hunk 6 above)
Comment (Slamdunk) [reply tests/MessageTest.php]: This is the change I was least comfortable with.

Even though `F1` is the right UTF-8 code for n with tilde, I was unable to reproduce in a real email the character written like this. Instead every mail agent uses =C3=B1.

I think we need to refactor test assets as a whole:

1. Write emails within PHP in pure UTF-8
1. Use an external tool/library/ext to convert them into different charsets
1. Send them to Dovecot (already done)
1. Test this library reads the result in UTF-8 (already done)

String concatenation seems too week to handle the encoding topic properly (in the tests).

Comment (ddeboer) [reply tests/MessageTest.php]: Fair enough. We can use ddeboer/transcoder (or some other lib) for step 2.

Comment (Slamdunk) [reply tests/MessageTest.php]: Reminder: https://github.com/ddeboer/imap/issues/176

It have to said that https://github.com/ddeboer/imap/commit/21f3817c674062d275537a7a818ba26e4e9962e9 behavied completely differently from now; something somewhere got broken and testing against Gmail didn't help.
